include/ccmath/math/nearest/rint.hpp

```diff
@@ -10,7 +10,239 @@
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr float rint(float num)
+	{
+		return ccm::rint<float>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr double rint(double num)
+	{
+		return ccm::rint<double>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long double rint(long double num)
+	{
+		return ccm::rint<long double>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr float rintf(float num)
+	{
+		return ccm::rint<float>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long double rintl(long double num)
+	{
+		return ccm::rint<long double>(num);
+	}
+
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value, using the current rounding mode. The library provides overloads of std::lrint and
+	 * std::llrint for all cv-unqualified floating-point types as the type of the parameter num.(
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	constexpr long lrint(float num)
+	{
+		return static_cast<long>(ccm::rint(num));
```

Comment:
Ok, this function here is a bit trickier and performing a `static_cast` won't actually be enough to respect all rounding modes. `static_cast` does not perform conversion with floats but instead performs truncation and this will always follow the default rounding mode. As such if the rounding mode is different than the default this should not work as expected. The proper solution is to instead get the bits of the float itself and perform rounding towards a signed integer using the bits themselve. There is a few manners to do this, but it will be a tricky problem to fix. Let me know your thoughts!

Reply:
I will have to investigate this after writing the tests I think, cause I initially assumed this would be an OK cast to perform since the long and float are the same size? I will have to brush up on my casting :)

Thanks for pointing this out!